.github/workflows/go-build-test.yml

```diff
@@ -15,30 +15,102 @@ on:
       - 'go.sum'
 
 jobs:
-  build-and-test:
+  build-and-test-multi-arch:
     runs-on: ubuntu-latest
+    name: Build on ${{ matrix.arch }}
+    # Run steps on a matrix of 2 arch.
+    strategy:
+      matrix:
+        arch:
+          - x64
+          - aarch64
     steps:
-      - uses: actions/checkout@v3
-
-      - name: Set up Go
-        uses: actions/setup-go@v3
-        with:
-          go-version: '1.18'
-          check-latest: true
-
-      - name: Build UI
-        run: make ui/build
-
-      - name: Test
-        run: make go/test
-
-      - name: Build
-        run: make build
-
-      - name: Archive generatated artifacts
-        uses: actions/upload-artifact@v3
-        with:
-          name: parca-bin
-          if-no-files-found: error
-          path: |
-            bin
+    - uses: actions/checkout@v3
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: '1.18'
+        check-latest: true
+
+    - name: Build UI
+      if: ${{ matrix.arch == 'x64' }}
+      run: |
+        make ui/build
+
+    - name: Test
+      if: ${{ matrix.arch == 'x64' }}
+      run: |
+        make go/test
+
+    - name: Build
+      if: ${{ matrix.arch == 'x64' }}
+      run: |
+        make build
+
+    - name: Archive generatated artifacts
+      if: ${{ matrix.arch == 'x64' }}
+      uses: actions/upload-artifact@v3
+      with:
+        name: parca-bin
+        if-no-files-found: error
+        path: |
+          bin
+
+    - name: 'Build and Test on ${{ matrix.arch }}'
+      if: ${{ matrix.arch != 'x64' }}
+      run: |
+        # Install yarn packages initially on AMD machine so that it can be mounted as volume
+        # on ARM docker container. This will save time as `yarn install` takes a lot of time
+        # on ARM container running using QEMU.
+        cd ui && yarn install && cd - 
+        export RUNNER_ALLOW_RUNASROOT="1"
+        ARCH=arm64
+
+        # Install QEMU and it's dependencies.
+        sudo apt-get install qemu binfmt-support qemu-user-static
+        docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+
+        # Run ARM based ubuntu image. Inside the container it installs Golang, which will be 
+        # used to build the code. Then it installs LTS version of node, which then 
+        # installs yarn to build UI workspace.
+        docker run --rm -t -v `pwd`:/parca arm64v8/ubuntu /bin/bash -c \
+        "uname -m &&\
+        cd parca &&\
+        # Update the container packages.
+        apt-get update -y -q &&\
+        apt-get upgrade -y -q &&\
+        apt-get install -y -q wget make git file build-essential &&\
+
+        # Install Golang.
```

Comment:
So the setup-go action doesn't handle multi arch?

Reply:
So how it works is we setup Go, but it is done inside host container  i.e github runner container which is  `linux/amd64`. But to test `linux/arm64`. We create a new docker image `arm64v8/ubuntu` and install all the necessary packages. For more information please refer to https://github.com/parca-dev/parca/issues/562#issuecomment-1129723914 and https://github.com/parca-dev/parca/issues/562#issuecomment-1129839656. I hope it makes sense.